File: Stock Market/China/A股/K线图绘制/A股数据.py
import plotly.graph_objects as go
import pandas as pd

# Read data from a csv
df = pd.read_excel('000001_平安银行.xlsx')

fig = go.Figure(data=[go.Candlestick(x=df['日期'],
                open=df['开盘'],
                high=df['最高'],
                low=df['最低'],
                close=df['收盘'])])

fig.show()

File: Stock Market/China/A股/K线图绘制/A股数据.R
# 加载所需库
library(plotly)

# 读取Excel文件，假设数据存储在'000001_平安银行.xlsx'中
library(readxl)
df <- read_excel('000001_平安银行.xlsx')

# 创建蜡烛图
fig <- df %>%
  plot_ly(x = ~日期, type="candlestick",
          open = ~开盘, close = ~收盘,
          high = ~最高, low = ~最低)

# 显示图表
fig
